openreview/venue/venue.py

```diff
@@ -1009,6 +1015,125 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
+                    "id": self.id,
+                    "name": self.name,
+                    "owners": [
+                    # {
+                    #     "id": "d7cf2650-c1c7-11e8-b568-0800200c9a66",
+                    #     "family_name": "test_instructor_first_name",
+                    #     "given_name": "test_instructor_last_name",
+                    #     "email": "instructor_email@test.com"
+                    # },
+                    # {
+                    #     "id": "7a62f070-c265-11e8-b568-0800200c9a66",
+                    #     "family_name": "test_instructor_2_first_name",
+                    #     "given_name": "test_instrutor_2_last_name",
+                    #     "email": "intructor_2_email@test.com"
+                    # }
+                    ]
+                },
+                group_type="ASSIGNMENT",
+            )
+            iThenticate_submission_id = res["id"]
+            
+
+            # Create a new edge with head as the note ID and tail as the iThenticate submission ID
+            iThenticate_edge = openreview.api.Edge(
+                invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+                head=submission.id,
+                tail=iThenticate_submission_id,
+                label="Created",
+                weight=-1,
+            )
+
+            
+            iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+            # Get the PDF for the submission
+            submission_file_binary_data = self.client.get_attachment(
+                id=submission.id, field_name="pdf"
+            )
+
+            submission_file_object = io.BytesIO(submission_file_binary_data)
+
+
+            # update the edge with status file sent
+            iThenticate_edge.label = "File Sent"
+            iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+
+            # Upload PDF to iThenticate
+            res = iThenticate_client.upload_submission(
+                submission_id=iThenticate_submission_id,
+                file_data=submission_file_object,
+                file_name=submission.content['title']['value']
+            )
+            
+
+    def ithenticate_request_similarity_report(self):
+        if not self.iThenticatePlagiarismCheck:
+            raise openreview.OpenReviewException('iThenticatePlagiarismCheck is not enabled for this venue.')
+        
+        iThenticate_client = openreview.api.iThenticateClient(self.iThenticatePlagiarismCheckApiKey, self.iThenticatePlagiarismCheckApiBaseUrl)
+
+        edges = self.client.get_grouped_edges(
+            invitation=self.get_iThenticate_plagiarism_check_invitation_id(), groupby='tail'
+        )
+
+        for edge in edges:
+            # change edge status to similarity requested
+            if edge["values"][0].label == "File Uploaded":
+                edge["values"][0].label = "Similarity Requested"
+                updated_edge = self.client.post_edge(edge["values"][0])
```

Comment:
again, post the edge after the operation is performed with no errors.

Reply:
I've added a try-except block that rolls back the label value in case of an error